drivers/frequency/hmc7044/hmc7044.c

```diff
@@ -290,6 +290,64 @@ uint32_t hmc7044_calc_out_div(uint32_t rate,
 	return div;
 }
 
+/**
+ * Recalculate rate corresponding to a channel.
+ * @param dev - The device structure.
+ * @param chan - Channel number.
+ * @param rate - Channel rate.
+ * @return SUCCESS in case of success, negative error code otherwise.
+ */
+uint32_t hmc7044_clk_recalc_rate(struct hmc7044_dev *dev, uint32_t chan,
+				 uint32_t *rate)
+{
+	if (chan > dev->num_channels)
+		return FAILURE;
+
+	*rate = dev->pll2_freq / dev->channels[chan].divider;
+
+	return SUCCESS;
+}
+
+/**
+ * Calculate closest possible rate
+ * @param rate - The desired rate.
+ * @param parent_rate - The parent rate.
+ * @return The closest possible rate of desired rate.
+ */
+uint32_t hmc7044_clk_round_rate(struct hmc7044_dev *dev, uint32_t rate,
+				uint32_t parent_rate)
+{
+	uint32_t div = hmc7044_calc_out_div(rate, parent_rate);
+
+	return DIV_ROUND_CLOSEST(dev->pll2_freq, div);
+}
+
+/**
+ * Set channel rate.
+ * @param dev - The device structure.
+ * @param chan - Channel number.
+ * @param rate - Channel rate.
+ * @return SUCCESS in case of success, negative error code otherwise.
+ */
+uint32_t hmc7044_clk_set_rate(struct hmc7044_dev *dev, uint32_t chan,
+			      uint32_t rate)
+{
+	uint32_t div;
+
+	if (chan >= dev->num_channels)
+		return FAILURE;
+
+	div = hmc7044_calc_out_div(rate, dev->pll2_freq);
+	dev->channels[chan].divider = div;
+
+	hmc7044_write(dev, HMC7044_REG_CH_OUT_CRTL_1(chan),
```

Comment:
can we check that the writes succeded and return that ?

ret = hmc7044_write

return ret ?